### PR TITLE
webtorrent: Add persistence

### PR DESCRIPTION
**Pull request by dodorz**

### CODE REVIEW

**Comment by linsui** [comment]
Please remove unrelated commit.

**Comment by linsui** [comment]
Please remove `#/dl.7z`, scoop will unpack nupkg.

**Comment by linsui** [comment]
Please copy config from appdata.

```
    "pre_install": [
        "if (!(Test-Path \"$persist_dir\\Portable Settings\")) {",
        "    New-Item \"$dir\\Portable Settings\" -ItemType 'Directory' -Force | Out-Null",
        "    foreach ($data in @(\"$env:Appdata\\WebTorrent\", \"$env:LocalAppdata\\WebTorrent\")) {",
        "        if (Test-Path $data) {",
        "            Write-Host -F yellow \"Copying old '$data' to '$persist_dir\\Portable Settings'\"",
        "            Copy-Item \"$data\\*\" \"$dir\\Portable Settings\" -Recurse -Force",
        "        }",
        "    }",
        "}"
    ],
```

**Comment by linsui** [comment]
We need copy old config in appdata. And if we only create a empty Portable Settings webtorrent can't find download directory.

```
    "pre_install": [
        "if (!(Test-Path \"$persist_dir\\Portable Settings\")) {",
        "    New-Item \"$dir\\Portable Settings\\Downloads\" -ItemType 'Directory' -Force | Out-Null",
        "    foreach ($data in @(\"$env:Appdata\\WebTorrent\", \"$env:LocalAppdata\\WebTorrent\")) {",
        "        if (Test-Path $data) {",
        "            Write-Host -F yellow \"Copying old '$data' to '$persist_dir\\Portable Settings'\"",
        "            Copy-Item \"$data\\*\" \"$dir\\Portable Settings\" -Recurse -Force",
        "        }",
        "    }",
        "}"
    ],
```

File: bucket/webtorrent.json [hunk 1]
```diff
@@ -5,15 +5,26 @@
...
-            "url": "https://github.com/webtorrent/webtorrent-desktop/releases/download/v0.20.0/WebTorrent-0.20.0-ia32-full.nupkg#/dl.7z",
+            "url": "https://github.com/webtorrent/webtorrent-desktop/releases/download/v0.20.0/WebTorrent-0.20.0-ia32-full.nupkg",
             "hash": "sha1:21B1CC158D4035DEEBCDC5B08DB8F2D4A37DC238"
         }
     },
     "extract_dir": "lib\\net45",
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\Portable Settings\")) {",
+        "    New-Item \"$dir\\Portable Settings\\Downloads\" -ItemType 'Directory' -Force | Out-Null",
+        "    foreach ($data in @(\"$env:Appdata\\WebTorrent\", \"$env:LocalAppdata\\WebTorrent\")) {",
```
Comment (Ash258) [inline bucket/webtorrent.json]: Is there a specific reason for both appdata?

Comment (linsui) [reply bucket/webtorrent.json]: There are different files in different directories.

Comment (Ash258) [inline bucket/webtorrent.json]: Copy variable is just for keeping copy item call shorter

```suggestion
        "    $copy = \"$env:LOCALAPPDATA\\Webtorrent\\*\", \"$env:APPDATA\\Webtorrent\\*\"",
        "    Copy-Item $copy \"$dir\\Portable Settings\" -ErrorAction SilentlyContinue -Force -Recurse",
```

File: bucket/webtorrent.json [hunk 2]
```diff
@@ -5,15 +5,27 @@
...
             "hash": "sha1:21B1CC158D4035DEEBCDC5B08DB8F2D4A37DC238"
         }
     },
     "extract_dir": "lib\\net45",
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\Portable Settings\")) {",
+        "    New-Item \"$dir\\Portable Settings\\Downloads\" -ItemType 'Directory' -Force | Out-Null",
+        "    $copy = \"$env:LOCALAPPDATA\\Webtorrent\\*\", \"$env:APPDATA\\Webtorrent\\*\"",
+        "    Copy-Item $copy \"$dir\\Portable Settings\" -ErrorAction SilentlyContinue -Force -Recurse",
+        "        if (Test-Path $data) {",
```
Comment (Ash258) [inline bucket/webtorrent.json]:
```suggestion
```

File: bucket/webtorrent.json [hunk 3]
```diff
@@ -5,15 +5,27 @@
...
+        "            Write-Host -F yellow \"Copying old '$data' to '$persist_dir\\Portable Settings'\"",
```
Comment (Ash258) [inline bucket/webtorrent.json]:
```suggestion
```

File: bucket/webtorrent.json [hunk 4]
```diff
@@ -5,15 +5,27 @@
...
+        "            Copy-Item \"$data\\*\" \"$dir\\Portable Settings\" -Recurse -Force",
```
Comment (Ash258) [inline bucket/webtorrent.json]:
```suggestion
```

File: bucket/webtorrent.json [hunk 5]
```diff
@@ -5,15 +5,26 @@
...
             "hash": "sha1:21B1CC158D4035DEEBCDC5B08DB8F2D4A37DC238"
         }
     },
     "extract_dir": "lib\\net45",
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\Portable Settings\")) {",
+        "    New-Item \"$dir\\Portable Settings\\Downloads\" -ItemType 'Directory' -Force | Out-Null",
+        "    $copy = \"$env:LOCALAPPDATA\\Webtorrent\\*\", \"$env:APPDATA\\Webtorrent\\*\"",
+        "    Copy-Item $copy \"$dir\\Portable Settings\" -ErrorAction SilentlyContinue -Force -Recurse",
+        "    if (Test-Path $data) {",
```
Comment (Ash258) [inline bucket/webtorrent.json]: Remove this whole `if` block

This block was needed before changing to copy-item with multiple sources.

**Comment by pratikpc** [comment]
Closing as dodorz:webtorrent has been deleted.

**Pull request closed without merging**